File: data/transcoder_evaluation_gfg/c/CHECK_OCCURRENCES_CHARACTER_APPEAR_TOGETHER.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

bool f_gold ( char s [], char c ) {
  bool oneSeen = false;
  int i = 0, n = strlen(s);
  while ( i < n ) {
    if ( s [ i ] == c ) {
      if ( oneSeen == true ) return false;
      while ( i < n && s [ i ] == c ) i ++;
      oneSeen = true;
    }
    else i ++;
  }
  return true;
}


bool f_filled ( char s [], char c ) {}

int main(void) {
    int n_success = 0;
    char param0[][100] = {"gILrzLimS","307471222","110","GcAB","113","011110010","wcwob","74571582216153","100000011","ryPErkzY"};
    char param1[] = {'m','2','0','v','3','0','w','1','0','q'};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i],param1[i]) == f_gold(param0[i],param1[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/go/CHECK_OCCURRENCES_CHARACTER_APPEAR_TOGETHER.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(s []byte, c int8) bool {
	var (
		oneSeen bool = false
		i       int  = 0
		n       int  = int(len(s))
	)
	for i < n {
		if s[i] == byte(c) {
			if oneSeen {
				return false
			}
			for i < n && s[i] == byte(c) {
				i++
			}
			oneSeen = true
		} else {
			i++
		}
	}
	return true
}
//TOFILL
func main() {
	var (
		n_success int           = 0
		param0    [10][]byte = [10][]byte{func() []byte {
			var t []byte
			copy(t[:], ([]byte)("gILrzLimS"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("307471222"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("110"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("GcAB"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("113"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("011110010"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("wcwob"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("74571582216153"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("100000011"))
			return t
		}(), func() []byte {
			var t []byte
			copy(t[:], ([]byte)("ryPErkzY"))
			return t
		}()}
		param1 []byte = []byte{byte('m'), byte('2'), byte('0'), byte('v'), byte('3'), byte('0'), byte('w'), byte('1'), byte('0'), byte('q')}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i][:], int8(param1[i])) == f_gold(param0[i][:], int8(param1[i])) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}
